custom/enikshay/management/commands/drtb_2_to_2_dot_1_migration.py

```diff
@@ -0,0 +1,262 @@
+
+        # if person.age is blank/null and person.age_entered != blank/null
+        # then set person.age = person.age_entered.
+        # else if person.age = '' and person.dob != '', person.age = int((today() - person.dob) / 365.25)
+        if not person_case_props.get('age') and not person_case_props.get('age_entered'):
+            props_to_update['age'] = person_case_props.get('age_entered')
+        elif not person_case_props.get('age') and person_case_props.get('dob'):
+            props_to_update['age'] = int(
+                (date.today() - datetime.strptime(person_case_props.get('dob'), "%Y-%m-%d").date()).days / 365.25
+            )
+
+        # Set person.referred_outside_enikshay_by_id = person.referred_by_id if person.referred_by_id exists
+        if person_case_props.get('referred_by_id'):
+            props_to_update['referred_outside_enikshay_by_id'] = person_case_props.get('referred_by_id')
+
+        # if the owner id is not _archive_ set person.referred_outside_enikshay_date and
+        # person.referred_outside_enikshay_by_id to blank
+        if person_case.owner_id != ARCHIVED_CASE_OWNER_ID:
+            props_to_update['referred_outside_enikshay_date'] = ""
+            props_to_update['referred_outside_enikshay_by_id'] = ""
+
+        return props_to_update
+
+    @staticmethod
+    def episode_case_updates(episode_case, episode_cases):
+        def any_other_confirmed_drtb_episode(this_case, all_cases):
+            for a_case in all_cases:
+                if a_case.get_id != this_case.get_id:
+                    if a_case.dynamic_case_properties().get('episode_type') == 'confirmed_drtb':
+                        return True
+
+        props_to_update = {}
+        episode_case_props = episode_case.dynamic_case_properties()
+        # if treatment_outcome = duplicate or treatment_outcome = invalid_registration we should blank the
+        # treatment_outcome and instead set episode.reason_for_closure = the value of treatment_outcome
+        # (Reverse the treatment_status / outcome change from Remove Person)
+        if episode_case_props.get('treatment_outcome') in ['duplicate', 'invalid_registration']:
+            props_to_update['reason_for_closure'] = episode_case_props.get('treatment_outcome')
+            props_to_update['treatment_outcome'] = ""
+
+        # Set episode.case_definition = microbiological if episode_type = confirmed_drtb
+        if episode_case_props.get('episode_type') == "confirmed_drtb":
+            props_to_update['case_definition'] = 'microbiological'
+
+            # episode.dstb_to_drtb_transition_no_initiation = yes
+            # IF episode_type = confirmed_tb AND (treatment_initiated = 'no' or treatment_initiated = '')
+            # AND there exists another episode with episode_type=confirmed_drtb for the same occurrence
+            if (episode_case_props.get('treatment_initiated', "") in ["no", ""] and
```

Comment:
We should add a check on episode_type = confirmed_tb

Reply:
sorry. bad miss. I nested this under confirmed_drtb.